frontend/fingertips-frontend/components/organisms/LineChart/index.tsx

```diff
@@ -17,28 +17,24 @@ export function LineChart({
   yAxisTitle,
   accessibilityLabel,
 }: Readonly<LineChartProps>) {
-  const categories = data.map(
-    (item) => item.date?.toLocaleDateString('en-GB') ?? ''
-  );
-  const temperatureData = data.map((item) => item.temperatureC ?? '');
+  const years = data.map((item) => item.healthData.map((point) => point.year));
+  const formatYear = years.flat().map((year) => year.toString());
 
   const lineChartOptions: Highcharts.Options = {
     chart: { type: 'line' },
     title: { text: title },
     xAxis: {
-      categories: categories,
+      categories: Array.from(new Set(formatYear)),
```

Comment:
How about a little helper function for this? something like `generateYearsForXAxis` where you do this and lines 20 and 21.